doc/filters/merge.rst

```diff
@@ -42,6 +42,25 @@ overridden.
         
 .. note::
 
-    Internally, Twig uses the PHP `array_merge`_ function.
+    Internally, Twig uses the PHP `array_merge`_ function, which means that it will loose any numerical index after a merge, unless the numerical index is concatenated with something to make a string.
```

Comment:
right, how would you phrase that so that it doesn't sound to convoluted.

You can keep the index by concatenating it to something but you cannot keep it numeric ( in one step ) ?

You can keep the index value by concatenating it to something to transform it into a string, but you can't keep the numeric character of the index.

...